test/gesture_detectors_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_hooks/src/framework.dart';
+import 'package:flutter_hooks/src/hooks.dart';
+
+import 'mock.dart';
+
+void _testGestureRecognizer<T>({
```

Comment:
Here is also one generic method that makes sure hook works as hooks should for all gesture detectors that would be tested here.

Reply:
I don't like the idea of having such abstraction.
I'd prefer removing it and make such that the expectations don't depend on anything

Reply:
Done!